Lib/test/test_type_aliases.py

```diff
@@ -0,0 +1,182 @@
+import types
+import unittest
+from test.support import check_syntax_error
+
+from typing import Callable, TypeAliasType, TypeVar, get_args
+
+from .test_type_params import run_code
```

Comment:
This import feels a bit weird to me. Would it be better to define the function in `test.support` somewhere? (I'm surprised there isn't a function like it already.)

Remind me why we have to run the tests that don't raise `SyntaxError`s as strings in the first place -- is it to ensure they're in an isolated scope?

Reply:
@erictraut wrote most of the tests that way to start with, so I just stuck with that. One advantage is that it catches refleaks in the compiler, because we run the compilation repeatedly.

And yes, I can move it to test.support.

Reply:
It also does not leak variables into the environment, which is amazing.

Reply:
Moved it to test.support (next to the similar `check_syntax_error`).

Reply:
Thanks!

Reply:
The other advantage is that when you break a specific construct used in just one or a few tests, you get a nice isolated failure of just those tests, rather than a failure to compile the entire test file that you then have to manually track down to the problematic lines.

And your tests are not affected by pyc file caching.

I generally think all tests intended to test the parser or compiler are better off if they dynamically compile strings.

Reply:
Makes sense, thanks!